x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.ts

```diff
@@ -124,14 +126,18 @@ export const endpointMiddlewareFactory: ImmutableMiddlewareFactory<EndpointState
     if (
       action.type === 'userChangedUrl' &&
       hasSelectedEndpoint(getState()) === true &&
-      getIsOnEndpointDetailsActivityLog(getState())
+      getIsOnEndpointDetailsActivityLog(getState()) &&
```

Comment:
could you turn all this complex condition check into a separate named function for ease of read?

Reply:
Indeed. Should be a selector here.

Reply:
done 66ea93868e98bf17839742f42bb9b8f71433a12a

Reply:
thanks!